Added functionality to 'convert_dates()'

nielsaka opened the pull request:
* can now handle quarterly data
* if date format unknown, returns input and prints warning

expersso commented:
Thanks, that's very useful! 

The travis build currently fails with `'::' or ':::' import not declared from: ‘zoo’`. 
I think it would probably be best to add `zoo` as a `Suggest` in `DESCRIPTION` and then test for it with `requireNamespace` in `convert_dates` (as suggested [here](http://stackoverflow.com/questions/15648772/how-do-i-prevent-r-library-or-require-calls-not-declared-warnings-when-dev)). 

Would you mind trying this?

nielsaka commented:
Sorry for taking so long to come back to this. 

Back in July I forgot to push the changes to `DESCRIPTION`, as I was under the misconception that roxygen would handle that part. `zoo` is now listed under `Suggests:` and the package namespace is loaded if required.

I dropped the `@importFrom` line, since [it's recommended](https://cran.r-project.org/web/packages/roxygen2/vignettes/namespace.html#imports) to either use `::` or `@importFrom` and also added a test.

Now there is a merge conflict (which I can't see...?), so CI does not seem to start. Locally, everything went fine for me.
